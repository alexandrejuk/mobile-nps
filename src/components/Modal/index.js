import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './style.module.css'

const Modal = ({
  children,
  close,
  show,
}) => (
  <div
    onClick={close}
    className={classNames(styles.bgModal, {
      [styles.show]: show,
    })}
  >
  <div
    className={classNames(styles.modal, {
      [styles.show]: show,
    })}
  >
      {children}
    </div>
   </div>
)

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  close: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
}

export default Modal
