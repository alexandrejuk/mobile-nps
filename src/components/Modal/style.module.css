@import '../../assets/css/colors.css';

.bgModal {
  align-items: center;
  background-color: rgba(0, 0, 0, 0.7);
  bottom: 0;
  box-sizing: border-box;
  color: var(--color-ship-gray);
  display: flex;
  flex-direction: column;
  justify-content: center;
  left: 0;
  opacity: 0;
  padding: 20px;
  position: fixed;
  right: 0;
  top: 0;
  transition: all 0.5s ease-out;
  visibility: hidden;
  z-index: 1;
}

.modal {
  background-color: var(--color-white);
  border-radius: 3px;
  box-sizing: border-box;
  font-size: 14px;
  max-width: 768px;
  padding: 20px;
  transform: translateY(-840px);
  transition: transform 0.5s ease 0s;
  width: 100%;
}

.show {
  transform: translateY(0);
}

.bgModal.show {
  opacity: 1;
  visibility: visible;
}
