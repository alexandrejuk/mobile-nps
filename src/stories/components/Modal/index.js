import React, { useState, Fragment } from 'react'
import { Button, Modal } from '../../../components'

export default {
  title: 'Components/Modal',
}

export const Default = () => {
  const [show, setShow] = useState(false)

  const openModal = () => setShow(true)
  const closeModal = () => setShow(false)

  return (
    <Fragment>
      <Button
        action={openModal}
        color="primary"
      >
        open modal
      </Button>
      <Modal
        close={closeModal}
        show={show}
      >
        <div>
          Your content here...
        </div>
        <Button
          action={closeModal}
          color="primary"
        >
          close modal
        </Button>
      </Modal>
    </Fragment>
  )
}
