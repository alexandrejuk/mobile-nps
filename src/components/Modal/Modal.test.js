import React from 'react'
import { render } from '@testing-library/react'

import Modal from './'

describe('Modal component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Modal
        show
        close={() => {}}
      >
        I am a modal component!
      </Modal>
    )
    expect(container).toMatchSnapshot()
  })
})
