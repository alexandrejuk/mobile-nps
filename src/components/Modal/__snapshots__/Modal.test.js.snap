// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render correctly 1`] = `
<div>
  <div
    class="bgModal show"
  >
    <div
      class="modal show"
    >
      I am a modal component!
    </div>
  </div>
</div>
`;
